nexus/src/app/background/sync_switch_configuration.rs

```diff
@@ -707,33 +694,54 @@ impl BackgroundTask for SwitchPortSettingsManager {
                     }
                 }).collect();
 
+                bgp.dedup();
+
                 let mut ports: Vec<PortConfigV1> = vec![];
 
                 for (location, port, change) in &changes {
                     let PortSettingsChange::Apply(info) = change else {
                         continue;
                     };
 
+                    let peer_configs = match self.datastore.bgp_peer_configs(opctx, *location, port.port_name.clone()).await {
```

Comment:
So this is where the primary fix is? Filtering the peers down based on switch location and port in this function which uses the new DB view?

Reply:
Yep. Instead of trying to build it from `SwitchPortSettingsCombinedResult`, we make a view where we query exactly what we want, for exactly what we're working on.

In hindsight, I should have taken this approach from the beginning for BGP configurations, and maybe even basic dendrite switch port settings. I'll definitely lean towards this in the future refactor.